sdk/openai/Azure.AI.OpenAI/src/Custom/ChatCompletionsOptions.Serialization.cs

```diff
@@ -174,6 +174,12 @@ void IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
                 writer.WritePropertyName("model"u8);
                 writer.WriteStringValue(InternalNonAzureModelName);
             }
+            if (AzureExtensionsOptions != null)
+            {
+                // CUSTOM CODE NOTE: Extensions options currently deserialize directly into the payload (not as a
+                //                      property value therein)
+                (AzureExtensionsOptions as IUtf8JsonSerializable).Write(writer);
```

Comment:
>(AzureExtensionsOptions as IUtf8JsonSerializable).Write(writer); [](http://example.com/codeflow?start=16&length=64)

Nitpick: I generally prefer a direct cast instead of using `as` in cases where we always expect it to be of a certain type:
`((IUtf8JsonSerializable)AzureExtensionsOptions).Write(writer);`
This way if something changes and the type of AzureExtensionOptions can't be converted IUtf8JsonSerializable, you will get a more helpful cast exception with a message like: `Unable to cast object of type 'NotFoo' to type 'IFoo'.`, instead of an unhelpful argument null exception